src/detext/run_detext.py

```diff
@@ -2,275 +2,148 @@
 Overall pipeline to train the model.  It parses arguments, and trains a DeText model.
 """
 
-import argparse
 import logging
 import os
 import sys
 import time
+from typing import NamedTuple, List
 
 import tensorflow as tf
 import tensorflow_ranking as tfr
+from arg_suite.arg_suite import arg_suite
 from detext.train import train
 from detext.train.data_fn import input_fn
 from detext.utils import misc_utils, logger, executor_utils
 
 
-def add_arguments(parser):
-    """Build ArgumentParser."""
-    parser.register("type", "bool", lambda v: v.lower() == "true")
-
+@arg_suite
+class Args(NamedTuple):
```

Comment:
What do we do if we want have mutually excluded group of arguments

Reply:
I don't think the mutually excluded group feature is a must, so not supported now. If you have a good use-case, please let me know.